virtualnetworks.go

```diff
@@ -11,6 +11,7 @@ type ProjectVirtualNetworkService interface {
 	List(projectID string, opts *ListOptions) (*VirtualNetworkListResponse, *Response, error)
 	Create(*VirtualNetworkCreateRequest) (*VirtualNetwork, *Response, error)
 	Get(string, *GetOptions) (*VirtualNetwork, *Response, error)
+	GetByVXLAN(string, int, *GetOptions) (*VirtualNetwork, *Response, error)
```

Comment:
Within a project, the same VXLAN (1000) could be used in different metros (representing different networks). This function would return one of these arbitrarily.

I was curious if `?vxlan=...&metro=...` might be available as filters, but they do not appear to be.

I think we could rename this to `GetByVXLANMetro` but this is not the sort of function we tend to add in packngo. Perhaps this helper could be added to the Terraform provider as `getVirtualNetworkByVXLANMetro`.

I have been curious about generic ways to add client-side filtering, https://github.com/packethost/packngo/issues/206.

Perhaps if `ListOptions` were an interface (`Optioner`?), we could offer getters, setters, and appenders, for ListOption fields and each implementation of `List` could type check for a specific implementation:
```go
func (i *ProjectVirtualNetworkServiceOp) List(projectID string, opts *ListOptions) (*VirtualNetworkListResponse, *Response, error) {
...
    if o, ok := opts.(*VirtualNetworkListOptions); ok {
      results = o.Filter(results)
    } // remove non-matches from response
```

```go
client.VLANs(projectID, 
  &packngo.VirtualNetworkListOptions{
    Filters: []func(*VirtualNetworkListResponse) bool{ ... }
  }
)
```

This could be an evolution of that approach (I'd prefer to make it even more generic):
```go
type Filterable interface {
  Filter()
}

// VirtualNetworkListWithFilters is admittedly a terrible name, probably owing to design
type VirtualNetworkListWithFilters {
  VirtualNetworks []VirtualNetwork
  Filters []func(VirtualNetwork) bool
}

// implement Filterable for VirtualNetworkListWithFilters, iterating VirtualNetworks against Filter funcs

type ListOption struct {
  ...
  Filter Filterable
}

func (i *ProjectVirtualNetworkServiceOp) List(projectID string, opts *ListOptions) (*VirtualNetworkListResponse, *Response, error) {
    ...
    
    if o, ok := opts.Filter.(* VirtualNetworkListWithFilters); ok {
      o.Filter.VirtualNetworks = append(o.Filter.VirtualNetworks, resp.VirtualNetworks)
      o.Filter.Filter()
      resp.VirtualNetworks = o.Filter.VirtualNetworks
    } // remove non-matches from response
    ...
}
```

```go
// specify filter functions in generic ListOptions with type specific filter functions via Filterable interface
client.VLANs(projectID, 
  &packngo.ListOptions{
    Filter: &packngo.VirtualNetworkListWithFilters(Filters: []func(packngo.VirtualNetwork) bool{ ... }
  }
)
```

Reply:
@displague I removed the ProjectVirtualNetworkServiceOp.GetByVXLAN. I only used it in Connection tests in packngo, and I just substituted it with a private helper.